### PR TITLE
 [Type Declaration] Skip AddArrayReturnDocTypeRector on return php doc

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #5028

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/type-declaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/skip_return_php_doc1.php.inc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+final class SkipReturnPhpDoc1
+{
+    /** @return array<int,array{a: string, b: float, c: \stdClass}> */
```
Comment (TomasVotruba) [inline rules/type-declaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/skip_return_php_doc1.php.inc:6]: What about nested cases?


```php
/** @return array<int, array<int, array{a: string, b: float, c: \stdClass}>> */
```

Comment (samsonasik) [reply rules/type-declaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/skip_return_php_doc1.php.inc:6]: I've added new fixture for nested case

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Great work, thank you :+1:

**Pull request merged**